src/components/hero.js

```diff
@@ -1,114 +1,251 @@
-      backgroundImage: `url("${heroImg}")`,
-      backgroundRepeat: "no-repeat",
-      backgroundPosition: "center",
-      backgroundSize: "cover",
-      width: "100%",
-      height: "100vh",
-      mt: "-105px",
-      mb: 4,
-    }}
-  >
+const getScaledHeroImage = (canvas, imageEl) => {
+  const heroImage = new fabric.Image(imageEl)
+  const canvasRatio = canvas.width / canvas.height
+  const imageRatio = imageEl.width / imageEl.height
+
+  if (imageRatio > canvasRatio) {
+    heroImage.scaleToHeight(canvas.height)
+  } else {
+    heroImage.scaleToWidth(canvas.width)
+  }
+
+  return heroImage
+}
+
+const rotatePath = (path, midpoint, { point, angle }) => {
+  const rotatePoint = fabric.util.rotatePoint(
+    point,
+    new fabric.Point(midpoint.x + point.x, midpoint.y + point.y),
+    fabric.util.degreesToRadians(angle)
+  )
+  path.set({
+    top: rotatePoint.y,
+    left: rotatePoint.x,
+    angle,
+  })
+}
+
+const drawRotatingBrandmark = (canvas, heroImageTemplate) => {
+  const heroImage = fabric.util.object.clone(heroImageTemplate)
+
+  const clipPath = new fabric.Path(brandmarkPath)
+  clipPath.absolutePositioned = true
+  clipPath.scale((1.2 * canvas.height) / clipPath.height)
+
+  const ANIMATION_TIME = 60 * 1000
+  const INITIAL_ANGLE = -15
+  const clipPathMidpoint = new fabric.Point(
+    500 * clipPath.getObjectScaling().scaleX,
+    581.9 * clipPath.getObjectScaling().scaleY
+  )
+  let startTimestamp = 0
+  let angle = INITIAL_ANGLE
+
+  const drawClip = timestamp => {
+    if (window.innerWidth === canvas.width) {
+      fabric.util.requestAnimFrame(drawClip)
+    } else {
+      canvas.clear()
+    }
+
+    rotatePath(clipPath, clipPathMidpoint, {
+      point: new fabric.Point(
+        (canvas.width / 4) * 3 - clipPath.getScaledWidth() / 2,
+        -(canvas.height / 4)
+      ),
+      angle,
+    })
+
+    const progress = (timestamp - startTimestamp) / ANIMATION_TIME
+    angle = -progress * 360 + INITIAL_ANGLE
+
+    canvas.remove(heroImage)
+    heroImage.clipPath = clipPath
+    canvas.add(heroImage)
+
+    heroImage.dirty = true
+    heroImage.center()
+  }
+  fabric.util.requestAnimFrame(timestamp => {
+    startTimestamp = timestamp
+    drawClip(timestamp)
+  })
+}
+
+const drawHeroBg = (canvas, imageEl) => {
+  const heroImage = getScaledHeroImage(canvas, imageEl)
+  drawRotatingBrandmark(canvas, heroImage)
+
+  const heroBg = fabric.util.object.clone(heroImage)
+  heroBg.set({ opacity: 0.2 })
+  canvas.add(heroBg)
+  heroBg.center()
+}
+
+const Hero = () => {
+  const canvasContainerRef = useRef(null)
+  const canvasRef = useRef(null)
+  const [imageEl, setImageEl] = useState(null)
+
+  const initializeCanvas = () => {
+    const { clientWidth, clientHeight } = canvasContainerRef.current
+    const canvas = new fabric.Canvas(canvasRef.current, {
+      width: clientWidth,
+      height: clientHeight,
+    })
+
+    return canvas
+  }
+
+  useEffect(() => {
+    const canvas = initializeCanvas()
```

Comment:
put initialize canvas declaration inside of useEffect. Why didn't this trigger a lint? 
If you are to use outside variables and functions, do add them as dependencies as a practice.
This is to ensure stableness because of javascript's inherent behavior of closures.

Reply:
> put initialize canvas declaration inside of useEffect.

`useEffect`? Do you mean `useCallback`?

> Why didn't this trigger a lint? If you are to use outside variables and functions, do add them as dependencies as a practice.

`initializeCanvas` is being recreated on every render so its reference inside `useEffect` here can't be ever stale.